storage/storage_helpers.cpp

```diff
@@ -28,28 +30,37 @@ bool IsEnoughSpaceForDownload(MwmSize mwmSize)
          Platform::TStorageStatus::STORAGE_OK;
 }
 
-bool IsEnoughSpaceForDownload(MwmSize mwmSizeDiff, MwmSize maxMwmSize)
-{
-  // Mwm size is less than |maxMwmSize|. In case of map update at first we download updated map
-  // and only after that we do delete the obsolete map. So in such a case we might need up to
-  // |maxMwmSize| of extra space.
-  return IsEnoughSpaceForDownload(mwmSizeDiff + maxMwmSize);
-}
-
 bool IsEnoughSpaceForDownload(CountryId const & countryId, Storage const & storage)
 {
   NodeAttrs nodeAttrs;
   storage.GetNodeAttrs(countryId, nodeAttrs);
+
   return IsEnoughSpaceForDownload(nodeAttrs.m_mwmSize);
 }
 
 bool IsEnoughSpaceForUpdate(CountryId const & countryId, Storage const & storage)
 {
   Storage::UpdateInfo updateInfo;
-  
   storage.GetUpdateInfo(countryId, updateInfo);
-  MwmSize spaceNeedForUpdate = updateInfo.m_sizeDifference > 0 ? updateInfo.m_sizeDifference : 0;
-  return IsEnoughSpaceForDownload(spaceNeedForUpdate, storage.GetMaxMwmSizeBytes());
+
+  /// @todo Review this logic when Storage::ApplyDiff will be restored.
+
+  // 1. For unlimited concurrent downloading process with "download and apply diff" strategy:
+  // - download and save all MWMs or Diffs = m_totalDownloadSizeInBytes
+  // - max MWM file size to apply diff patch (patches are applying one-by-one) = m_maxFileSizeInBytes
+  // - final size difference between old and new MWMs = m_sizeDifference
+
+  MwmSize const diff = updateInfo.m_sizeDifference > 0 ? updateInfo.m_sizeDifference : 0;
+  //return IsEnoughSpaceForDownload(diff + updateInfo.m_totalDownloadSizeInBytes + updateInfo.m_maxFileSizeInBytes);
```

Comment:
Потому что до применения диффа он может занимать место, зарезервированное под разницу размеров new-old. А после применения дифф удаляется всё равно.
```suggestion
  //return IsEnoughSpaceForDownload(std::max(diff, updateInfo.m_totalDownloadSizeInBytes) + updateInfo.m_maxFileSizeInBytes);
```

Reply:
Вроде нет ибо в моменте (перед удалением дифа) на диске будет лежать:
diff (от предидущих mwm)  + downloaded diff + old mwm + new mwm

Reply:
в смысле скачанные диффы удаляются только все вместе в конце, а не каждый индивидуально сразу после применения?

Если индивидуально, то для каждой конкретной карты сразу после применения диффа: old mwm + diff (лежит в резерве max(sizeDiff, totalDownload)) + new mwm (лежит в резерве maxFileSize).
Сразу после этого дифф и старый мвм удаляются и обрабатывается новый дифф.
Скачанные, но не примененные диффы от других мвм не мешают, потому что резерв max(sizeDiff, totalDownload) это же сумма резервов для каждой мвм, т.е. де-факто шарится для обработки между мвм будет только резерв maxFileSize.